### DISCUSSION TITLE
Can't get tailwind 3 to work with next.js inside of a nx monorepo

### DISCUSSION THREAD

**Discussion by danerwilliams** [tailwindlabs/tailwindcss#7446] [Help]
```
"tailwindcss": "^3.0.22",
"next": "12.0.7",
"postcss": "^8.4.6",
"autoprefixer": "^10.4.2",
```

Been trying to use [this tutorial](https://blog.nrwl.io/setup-next-js-to-use-tailwind-with-nx-849b7e21d8d0) to set up tailwind css in a next.js project that's part of an [nx monorepo](https://nx.dev/). 

It seems like this guide was written for tailwind 2.0, but I've tried looking through the [upgrade guide](https://tailwindcss.com/docs/upgrade-guide) and still haven't gotten it figured out. 

It appears that the tailwind styling just fails to implement entirely:
![Screen Shot 2022-02-13 at 3 06 53 PM](https://user-images.githubusercontent.com/22798229/153772789-e530fa76-23d1-4d35-95b6-fa02a0b5ec7c.png)

With the following setup:

`monorepo/apps/frontend/pages/_app.tsx`
```
import { AppProps } from 'next/app';

import 'tailwindcss/tailwind.css';

function CustomApp({ Component, pageProps }: AppProps) {
  return (
    <>
      <main className="app">
        <Component {...pageProps} />
      </main>
    </>
  );
}

export default CustomApp;
```
I've also tried importing from `styles.css` and setting `styles.css` with the `@tailwind` directives and got the same result of no styling on the page.  

`monorepo/apps/frontend/postcss.config.js`
```
module.exports = {
  plugins: {
    tailwindcss: {},
    autoprefixer: {},
  },
};
```
While I understand that if nothing is specified it should automatically point to `tailwind.config.js`, i've also tried setting the config property of `tailwindcss` to `apps/frontend/tailwind.config.js` and `join(__dirname, 'tailwind.config.js')` with the same result.

`monorepo/apps/frontend/tailwind.config.js`
```
const { createGlobPatternsForDependencies } = require('@nrwl/react/tailwind');
const { join } = require('path');

module.exports = {
  content: [''pages/**/*.{js,ts,jsx,tsx}''],
  theme: {
    extend: {},
  },
  plugins: [],
};
```
I've tried setting this to many other values as well. [This issue](https://github.com/nrwl/nx/issues/8355#issuecomment-1004690204) seemed to be resolved by setting the content based on the root of the monorepo, but this approach didn't work for me :-( either.

`monorepo/apps/frontend/pages/index.tsx`
```
export function Index() {
  return (
    <div className="bg-gray-50">
      <div className="max-w-7xl mx-auto py-12 px-4 sm:px-6 lg:py-16 lg:px-8 lg:flex lg:items-center lg:justify-between">
        <h2 className="text-3xl font-extrabold tracking-tight text-gray-900 sm:text-4xl">
          <span className="block">Ready to dive in?</span>
          <span className="block text-indigo-600">
            Start your free trial today.
          </span>
        </h2>
        <div className="mt-8 flex lg:mt-0 lg:flex-shrink-0">
          <div className="inline-flex rounded-md shadow">
            <a
              href="#"
              className="inline-flex items-center justify-center px-5 py-3 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700"
            >
              Get started
            </a>
          </div>
          <div className="ml-3 inline-flex rounded-md shadow">
            <a
              href="#"
              className="inline-flex items-center justify-center px-5 py-3 border border-transparent text-base font-medium rounded-md text-indigo-600 bg-white hover:bg-indigo-50"
            >
              Learn more
            </a>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Index;
```

Anyone have ideas on how to get this to work? I had no problems upgrading from 2 to 3 when not inside an nx monorepo so I'm not sure what to think. 

I can get it to work fine with tailwind 2:
```
"tailwindcss": "^2.2.19",
"next": "12.0.7",
"postcss": "^8.4.3",
"autoprefixer": "^10.4.0",
```
![Screen Shot 2022-02-13 at 3 47 17 PM](https://user-images.githubusercontent.com/22798229/153774373-d4d4fee0-d131-4ea4-ac49-39b9ed1043f1.png)

**Comment by vizardkill** [comment]
up

**Comment by bernaferrari** [comment]
I had the same issue and I did this:
<img width="539" alt="image" src="https://user-images.githubusercontent.com/351125/225807131-25f3eb11-2e05-43ac-93ea-5a66dde68ddd.png">
